polls/views.py

```diff
@@ -364,6 +365,57 @@ def test_func(self):
         return self.request.user.is_superuser
 
 
+class TemporaryLogin(View):
+    template_name = "polls/temp_login_result.html"
+
+    def post(self, request):
+        context = {"message": "You have been logged in temporarily"}
+        query_email = request.POST.get("email", None)
+        username = query_email.split("@")[0]
+        temp_password = "Asdfghjkl123"
+        try:
+            temp_user = get_user_model().objects.create(
+                username=username, email=query_email
+            )
+            temp_user.set_password(temp_password)
+            temp_user.save()
+            user = authenticate(request, username=username, password=temp_password)
+            login(request, user)
+        except IntegrityError:
+            context["message"] = "A user with this email already exists"
+        return render(request, self.template_name, context)
+
+
+class TemporaryLoginTemplate(TemplateView):
+    template_name = "polls/temp_login_template.html"
+
+
+class CreateExperimentTypeAnnotationView(LoginRequiredMixin, UserPassesTestMixin, View):
+    def get(self, request):
+        experiment_types = ExperimentType.objects.all()
+        context = {"experiment_types": experiment_types}
+        return render(request, "polls/experiment-type-annotation.html", context)
+
+    def post(self, request, *args, **kwargs):
+        exp_type = request.POST.get("experiment-type")
+        annotation = request.POST.get("annotation")
+        experiment_type = ExperimentType.objects.filter(pk=exp_type)
```

Comment:
I guess using "get" instead of "filter" will be a better choice as we are only getting a single object.

Reply:
The primary purpose of querying db is to ensure an instant of experiment-type in the db, and I can check for it only with the `.exists()`, `.exists()` works only with the `.filter` method.
Alternatively, I could have used `get` again and check if the variable is empty or not, but Django would automatically return 500 if get fails to get an instance in the db.
So, the only possible way I found is `.filter().exists()`.

Reply:
Actually I was thinking that we are using a dropdown for experiment types so we will always get the experiment using get. So we might not need an if check for that